californium-core/src/main/java/org/eclipse/californium/core/network/EndpointManager.java

```diff
@@ -94,6 +95,36 @@ public static EndpointManager getEndpointManager() {
 	private Endpoint default_secure_tpc_endpoint;
 
 	/**
+	 * Get default endpoint for provided scheme.
+	 * 
+	 * @param scheme scheme to select endpoint
+	 * @return the default endpoint for the provided scheme
+
+	 * @see #getDefaultEndpoint()
+	 * @see #getDefaultSecureEndpoint()
+	 * @see #getDefaultTcpEndpoint()
+	 * @see #getDefaultSecureTcpEndpoint()
+	 */
+	public Endpoint getDefaultEndpoint(String scheme) {
```

Comment:
I'd rather keep the *default* endpoints in a `Map<String, Endpoint>` with the *scheme* as keys so that we can add more endpoints and do not need to do the string comparisons.

I also think that the *Default* should be dropped form all of `EndpointManager`'s methods. The whole purpose of the `EndpointManager` is to deal with the default endpoints, no need to explicitly mention this in the method names again. And we are already inconsistent there as well, e.g. `setSecureTcpEndpoint()` ...

We probably do not need the `getDefault???Endpoint()` methods any more, do we? We could keep a `getEndpoint()` method returning the standard coap endpoint but otherwise let client code use a `getEndpoint(String scheme)` to get an endpoint for the given scheme.

WDYT?